### PR TITLE
[wb1812.3.migratewb] Migrate Wonder Blocks off old id providers

### PR DESCRIPTION
**Pull request by somewhatabstract**
## Summary:
This is the last piece in the first batch of work. This migrates all Wonder Blocks components off our old ID providers and onto the new `Id` component.

There are also some documentation tweaks to make the deprecation clearer in our stories, since that's some primary documentation for folks.

With this PR, we can cut a release and then begin updating consumer repos accordingly. First, to include this update, then to migrate them off the old ways and onto the new.

### Release process:
Once this small stack of PRs are landed and released, the following PRs need to be updated with that release, then landed/deployed:

- Perseus: https://github.com/Khan/perseus/pull/2007
- Webapp
  - https://github.com/Khan/webapp/pull/28105
  - https://github.com/Khan/webapp/pull/28127

Issue: WB-1812

## Test plan:
`yarn test`
`yarn typecheck`
`yarn start:storybook`

### CODE REVIEW

File: packages/wonder-blocks-dropdown/src/components/multi-select.tsx [hunk 1]
```diff
@@ -546,7 +545,7 @@ const MultiSelect = (props: Props) => {
         const menuText = getMenuText(allChildren);
 
         const dropdownOpener = (
-            <IDProvider id={id} scope="multi-select-opener">
+            <Id id={id}>
```
Comment (jandrade) [inline packages/wonder-blocks-dropdown/src/components/multi-select.tsx:548]: **nit:** This is now a function component, so you could use `useId` directly as well. I understand if you want to keep things simple for this migration.

File: packages/wonder-blocks-dropdown/src/components/single-select.tsx [hunk 2]
```diff
@@ -444,7 +443,7 @@ const SingleSelect = (props: Props) => {
             : placeholder;
 
         const dropdownOpener = (
-            <IDProvider id={id} scope="single-select-opener">
+            <Id id={id}>
```
Comment (jandrade) [inline packages/wonder-blocks-dropdown/src/components/single-select.tsx:446]: Same suggestion about `useId` (no changes required).

**Review by jandrade — APPROVED** [review]
Looks great to me! thanks for the quick refactor :shipit:

File: packages/wonder-blocks-dropdown/src/components/multi-select.tsx (hunk 1 above)
Comment (somewhatabstract) [reply packages/wonder-blocks-dropdown/src/components/multi-select.tsx:548]: Yeah, I considered that, but the component seems so complex; this felt cleaner and a little easier. :shrug:

Comment (jandrade) [reply packages/wonder-blocks-dropdown/src/components/multi-select.tsx:548]: makes total sense :)

File: packages/wonder-blocks-dropdown/src/components/combobox.tsx [hunk 3]
```diff
@@ -684,7 +680,6 @@ export default function Combobox({
                                         testId ? `${testId}-listbox` : undefined
                                     }
                                     aria-label={labels.listbox}
-                                    aria-labelledby={ids.get("input")}
```
Comment (somewhatabstract) [inline packages/wonder-blocks-dropdown/src/components/combobox.tsx:687]: **question:** @jandrade seems like this was a bug? There's already an `aria-label` attribute, and setting `aria-labelledby` to an ID that isn't attached to any element feels like it'll just confuse screen reader tech and browsers.

Comment (somewhatabstract) [reply packages/wonder-blocks-dropdown/src/components/combobox.tsx:687]: For the record, there was no bug here, I don't think. We're using label and labelledby, which may be an oversight, but it should be fine as-is.

**Pull request merged**